### PR TITLE
[FIX] base: do not fetch computed field already in cache

### PR DESCRIPTION
**Pull request by VincentSchippefilt**
Before this commit if a computed field is already in cache, but not its dependencies, `fetch` would fetch those dependencies. This commit ensures that fetch checks first if a computed is in cache before fetching its dependencies.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/addons/test_performance/tests/test_performance.py [hunk 1]
```diff
@@ -46,6 +46,7 @@ def setUpClass(cls):
             'value': 40,
             'partner_id': partner12.id,
         }])
+        cls.env.invalidate_all()
```
Comment (rco-odoo) [inline odoo/addons/test_performance/tests/test_performance.py:49]: Why?

File: odoo/addons/test_performance/tests/test_performance.py [hunk 2]
```diff
@@ -145,6 +146,20 @@ def test_fetch(self):
             # 'display_name' depends on name that should already be fetched
             records.fetch(['id', 'display_name'])
 
+        with self.assertQueryCount(0):
+            # touch the field display_name, they are now in cache
+            records.mapped('display_name')
+
+            # remove the dependencies of display_name (only name)
+            self.env.cache.remove(records[0], records._fields["name"])
```
Comment (rco-odoo) [inline odoo/addons/test_performance/tests/test_performance.py:154]:
```suggestion
            records.invalidate_recordset(['name'])
```

**Comment by VincentSchippefilt** [comment]
robodoo r+

**Comment by rco-odoo** [comment]
@VincentSchippefilt while resolving a conflict in the forward-port, I realized that we have dropped the call to `check_field_access_rights()` in this pull request.  We should fix it, and adapt the forward-port accordingly.

File: odoo/models.py [hunk 3]
```diff
@@ -3235,30 +3221,7 @@ def fetch(self, field_names):
         if not self or not field_names:
             return
 
-        # determine fields to fetch
-        fields_to_verify = OrderedSet()
-        field_names = self.check_field_access_rights('read', field_names)
```
Comment (rco-odoo) [inline odoo/models.py:3240]: We dropped that call in both `search_fetch()` and `fetch()`.
Either we reintroduce it in both methods, or we move it inside `_determine_fields_to_fetch()`.

**Pull request closed without merging**